src/code/audio/code_801A5BD0.c

```diff
@@ -1,45 +1,780 @@
+                Audio_QueueCmdS8(0x06020000 | ((entry->channelIdx & 0xFF) << 8), 0);
+            }
+            if (entry->state != SFX_STATE_EMPTY) {
+                Audio_RemoveSfxBankEntry(bankId, entryIndex);
+            }
+        } else {
+            prevEntryIndex = entryIndex;
+        }
+
+        entryIndex = gSfxBanks[bankId][prevEntryIndex].next;
+    }
+}
+
+void Audio_StopSfxByPosAndBank(u8 bankId, Vec3f* pos) {
+    SfxBankEntry entryToRemove;
+
+    func_801A7168(bankId, pos);
+    entryToRemove.sfxId = bankId << 12;
+    entryToRemove.posX = &pos->x;
+    Audio_RemoveMatchingSfxRequests(1, &entryToRemove);
+}
+
+void Audio_StopSfxByPos(Vec3f* pos) {
+    u8 bankId;
+    SfxBankEntry entryToRemove;
+
+    for (bankId = 0; bankId < ARRAY_COUNT(gSfxBanks); bankId++) {
+        func_801A7168(bankId, pos);
+    }
+
+    entryToRemove.posX = &pos->x;
+    Audio_RemoveMatchingSfxRequests(2, &entryToRemove);
+}
+
+void Audio_StopSfxByPosAndId(Vec3f* pos, u16 sfxId) {
+    SfxBankEntry* entry;
+    u8 entryIndex = gSfxBanks[SFX_BANK(sfxId)][0].next;
+    u8 prevEntryIndex = 0;
+    SfxBankEntry entryToRemove;
+
+    while (entryIndex != 0xFF) {
+        entry = &gSfxBanks[SFX_BANK(sfxId)][entryIndex];
+        if (entry->posX == &pos->x && entry->sfxId == sfxId) {
+            if (entry->state >= SFX_STATE_PLAYING_REFRESH) {
+                Audio_QueueCmdS8(0x06020000 | ((entry->channelIdx & 0xFF) << 8), 0);
+            }
+            if (entry->state != SFX_STATE_EMPTY) {
+                Audio_RemoveSfxBankEntry(SFX_BANK(sfxId), entryIndex);
+            }
+            entryIndex = 0xFF;
+        } else {
+            prevEntryIndex = entryIndex;
+        }
+
+        if (entryIndex != 0xFF) {
+            entryIndex = gSfxBanks[SFX_BANK(sfxId)][prevEntryIndex].next;
+        }
+    }
+
+    entryToRemove.posX = &pos->x;
+    entryToRemove.sfxId = sfxId;
+    Audio_RemoveMatchingSfxRequests(3, &entryToRemove);
+}
+
+void Audio_StopSfxByTokenAndId(u8 token, u16 sfxId) {
+    SfxBankEntry* entry;
+    u8 entryIndex = gSfxBanks[SFX_BANK(sfxId)][0].next;
+    u8 prevEntryIndex = 0;
+    SfxBankEntry entryToRemove;
+
+    while (entryIndex != 0xFF) {
+        entry = &gSfxBanks[SFX_BANK(sfxId)][entryIndex];
+        if (entry->token == token && entry->sfxId == sfxId) {
+            if (entry->state >= SFX_STATE_PLAYING_REFRESH) {
+                Audio_QueueCmdS8(0x06020000 | ((entry->channelIdx & 0xFF) << 8), 0);
+            }
+            if (entry->state != SFX_STATE_EMPTY) {
+                Audio_RemoveSfxBankEntry(SFX_BANK(sfxId), entryIndex);
+            }
+        } else {
+            prevEntryIndex = entryIndex;
+        }
+
+        if (entryIndex != 0xFF) {
+            entryIndex = gSfxBanks[SFX_BANK(sfxId)][prevEntryIndex].next;
+        }
+    }
+
+    entryToRemove.token = token;
+    entryToRemove.sfxId = sfxId;
+    Audio_RemoveMatchingSfxRequests(4, &entryToRemove);
+}
+
+void Audio_StopSfxById(u32 sfxId) {
```

Comment:
Hmm, I see that "by" makes more sense when you have all these laid out like this... I wish there was a better way to say "with specified X (and Y).

Reply:
`By` -> `Using`?        `Audio_StopSfxUsingPosAndId`?
`By` -> `With`           `Audio_StopSfxWithPosAndId`?
`By` -> `Through`     `Audio_StopSfxThroughPosAndId`?

Reply:
Of those I like `With` the best, although it's still not perfect. `Through` doesn't sound right to me, and `Using` suggests that the properties are more involved than just being a category type thing.